routes/project/location/kubernetes_cluster.rb

```diff
@@ -2,44 +2,54 @@
 
 class Clover
   hash_branch(:project_location_prefix, "kubernetes-cluster") do |r|
-    r.web do
-      r.on KUBERNETES_CLUSTER_NAME_OR_UBID do |kc_name, kc_ubid|
-        filter = if kc_name
-          {Sequel[:kubernetes_cluster][:name] => kc_name}
-        else
-          {Sequel[:kubernetes_cluster][:id] => UBID.to_uuid(kc_ubid)}
+    r.get api? do
+      kubernetes_cluster_list
+    end
+
+    r.on KUBERNETES_CLUSTER_NAME_OR_UBID do |kc_name, kc_ubid|
+      filter = if kc_name
+        r.post api? do
+          check_visible_location
+          kubernetes_cluster_post(kc_name)
         end
 
-        filter[:location_id] = @location.id
-        kc = @project.kubernetes_clusters_dataset.first(filter)
+        {Sequel[:kubernetes_cluster][:name] => kc_name}
+      else
+        {Sequel[:kubernetes_cluster][:id] => UBID.to_uuid(kc_ubid)}
+      end
 
-        next 404 unless kc
+      filter[:location_id] = @location.id
+      kc = @project.kubernetes_clusters_dataset.first(filter)
 
-        r.is do
-          r.get do
-            authorize("KubernetesCluster:view", kc.id)
+      next 404 unless kc
 
+      r.is do
+        r.get true do
```

Comment:
You shouldn't be adding the `true` argument here, you are already inside an `r.is` block, so the path has already been full routed.  Same with the following `r.delete` call.